docs/repository.rst

```diff
@@ -0,0 +1,21 @@
+PyTeal Repository Structure
```

Comment:
I think a question the PR prompts is:  Do we want to apply the same docs model for contributors as users?

* The user guide is generated from rst files and lands on https://pyteal.readthedocs.io/en/stable/.  Existing contributor docs are markdown local to the repo (e.g. https://github.com/algorand/pyteal/blob/master/CONTRIBUTING.md).
* I have an opinion, but I don't feel strongly and am interested in discussion:
  * Particularly because PyTeal is an open source project, it seems _reasonable_ to me that contributor docs ought to follow the user guide model.
  * While I think consistency is worth considering, I also don't believe _every_ internal doc ought to become an rst.  For example, I favor markdown files local to specific directories for providing context.

Reply:
Since we already have the bulk of our documents in RST, it made sense to me to continue building these out. But like yourself, I'm not super opinionated about it. I do dread the thought of having two different documentation systems with two different standards in the same repo. But a one-off doc here and there is certainly fine.